src/Engines/QueryBuilderEngine.php

```diff
@@ -394,7 +399,28 @@ private function getSearchKeyword($i, $raw = false)
 
         return $this->setupKeyword($keyword);
     }
-
+    
+    /**
+     * Check if model use SoftDeletes trait
+     *
+     * @return boolean
+     */
+    private function modelUseSoftDeletes()
+    {
+        $app = app();
+        
+        //check for laravel version
+        //in 4 and 5 versions SoftDeletes trait has a different names
+        //we can check only first number
+        if (strpos($app::VERSION, '5') === 0) {
+            $class = 'Illuminate\Database\Eloquent\SoftDeletes';
+        } else {
+            $class = 'Illuminate\Database\Eloquent\SoftDeletingTrait';
+        }
+        
+        return in_array($class, class_uses($this->query->getModel()));
```

Comment:
In fact, this method(class_uses) is created for this(check of triat usage). I try to find other methods, but in reality it is not so much. Anoter way is check of method_exists(), but i think this is worse. And one more way, do this, is use of interfaces(contracts) but for the package is not suitable. 
The only thing that we can fix it use instans of class SoftDeletes
in_array(new SoftDeletes, class_uses($this->query->getModel()))
but it will create a new object, which essentially would only take a memory